packages/financial-templates-lib/src/price-feed/HarvestVaultPriceFeed.js

```diff
@@ -0,0 +1,114 @@
+const { PriceFeedInterface } = require("./PriceFeedInterface");
+const { BlockFinder } = require("./utils");
+const { ConvertDecimals } = require("@uma/common");
+const assert = require("assert");
+
+class HarvestVaultPriceFeed extends PriceFeedInterface {
```

Comment:
Since this is mostly copied from the VaultPriceFeed, might I propose a way to make this a derived of that one?

```js
class VaultPriceFeedBase {
...
    const underlyingTokenAddress = await this._tokenTransaction().call();
...

  _tokenTransaction() {
    throw new Error("Must be implemented by derived class");
  }
}

// Note: we may rename this in the future to YearnV1Vault or something, but just for simplicity, we can keep the name the same for now.
class VaultPriceFeed extends VaultPriceFeedBase {
  _tokenTransaction() {
    return this.vault.methods.token();
  }
}

class HarvestVaultPriceFeed extends VaultPriceFeedBase {
  _tokenTransaction() {
    return this.vault.methods.underlying();
  }
}
```

You could even put them in the same file if you like. We do something similar in the UniswapPriceFeed!

The way you have it now also works, it's just difficult to tell what's new code and what has changed. From your description, the differences seem pretty small!

Reply:
Will update